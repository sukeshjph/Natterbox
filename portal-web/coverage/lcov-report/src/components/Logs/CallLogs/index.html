
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/components/Logs/CallLogs</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../../index.html">All files</a> src/components/Logs/CallLogs</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">12.82% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>10/78</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">2.56% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>1/39</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">12% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>3/25</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">12.5% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>9/72</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file empty" data-value="CallLogColDef.ts"><a href="CallLogColDef.ts.html">CallLogColDef.ts</a></td>
	<td data-value="0" class="pic empty">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file low" data-value="CallLogs.tsx"><a href="CallLogs.tsx.html">CallLogs.tsx</a></td>
	<td data-value="5.71" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 5%"></div><div class="cover-empty" style="width: 95%"></div></div>
	</td>
	<td data-value="5.71" class="pct low">5.71%</td>
	<td data-value="70" class="abs low">4/70</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="37" class="abs low">0/37</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="21" class="abs low">0/21</td>
	<td data-value="6.15" class="pct low">6.15%</td>
	<td data-value="65" class="abs low">4/65</td>
	</tr>

<tr>
	<td class="file low" data-value="CallLogsData.ts"><a href="CallLogsData.ts.html">CallLogsData.ts</a></td>
	<td data-value="0" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	</tr>

<tr>
	<td class="file high" data-value="CallLogsHeader.tsx"><a href="CallLogsHeader.tsx.html">CallLogsHeader.tsx</a></td>
	<td data-value="85.71" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 85%"></div><div class="cover-empty" style="width: 15%"></div></div>
	</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Wed Apr 01 2020 17:26:30 GMT+0100 (British Summer Time)
            </div>
        </div>
        <script src="../../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../../sorter.js"></script>
        <script src="../../../../block-navigation.js"></script>
    </body>
</html>
    