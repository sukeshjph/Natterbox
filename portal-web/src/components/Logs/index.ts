export { CallLogs } from "./components/CallLogs/CallLogs"
export { EventLogs } from "./components/EventLogs"
export { BulkDownloads } from "./components/BulkDownloads"
export { MyCalls } from "./components/MyCalls"
export { SMSMMSLogs } from "./components/SMSMMSLogs"
