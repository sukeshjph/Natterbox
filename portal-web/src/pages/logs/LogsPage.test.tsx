import React from "react"
import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import Logs from "./LogsPage"

describe("Logs", () => {
  const LogProps = {
    children: <div>Any react children</div>,
  }
  const wrapper = shallow(<Logs {...LogProps} />)

  it("Renders Logs", () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it.skip("should select Call Logs tab by default", () => {
    expect(wrapper.find(".tabsMenu").prop("value")).toBe(0)
  })

  describe("should render given react component for selected tab", () => {
    it("selected tab children", () => {
      const newLogProps = {
        children: <div id="child1">Updated children</div>,
      }
      wrapper.setProps({ ...newLogProps })
      expect(wrapper.find("#child1")).toBeDefined()
    })
  })
})
