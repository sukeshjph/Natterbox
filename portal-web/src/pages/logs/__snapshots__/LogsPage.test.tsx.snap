// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logs Renders Logs 1`] = `
<Memo()
  TabsMenu={
    Object {
      "Bulk Downloads": <BulkDownloads />,
      "Call Logs": <CallLogs />,
      "Event Logs": <EventLogs />,
      "My Calls": <MyCalls />,
      "SMS/MMS Logs": <SMSMMSLogs />,
    }
  }
/>
`;
